stdlib/hashtbl.ml

```diff
@@ -583,20 +604,28 @@ let find_all h key =
 
 let rec replace_bucket key data = function
   | Empty ->
-      true
+      None
   | Cons ({key=k; next} as slot) ->
       if compare k key = 0
-      then (slot.key <- key; slot.data <- data; false)
+      then (
+        let old_data = slot.data in
+        slot.key <- key;
+        slot.data <- data;
+        Some old_data)
       else replace_bucket key data next
 
-let replace h key data =
+let find_and_replace h key data =
   let i = key_index h key in
   let l = h.data.(i) in
-  if replace_bucket key data l then begin
-    h.data.(i) <- Cons{key; data; next=l};
-    h.size <- h.size + 1;
-    if h.size > Array.length h.data lsl 1 then resize key_index h
-  end
+  match replace_bucket key data l with
+      | None ->
+          h.data.(i) <- Cons{key; data; next=l};
+          h.size <- h.size + 1;
+          if h.size > Array.length h.data lsl 1 then resize key_index h;
+          None
+      | (Some _) as p -> p
```

Comment:
For uniformity, this code should be formatted exactly the same as the one in 453.